### PR TITLE
fix: actions docs

### PR DESCRIPTION
**Pull request by abhishekpatil4**
> [!IMPORTANT]
> Enhances Composio documentation with examples, sections for executing, creating, and finding actions, and restructures files for clarity.
> 
>   - **Documentation Enhancements**:
>     - Added examples for selecting actions by ID and filtering by tags in `action-guide-faqs.mdx`.
>     - Introduced a section on getting parameters for actions in `action-guide-without-agents.mdx`.
>     - Added examples for executing actions with and without agents in `action-guide-with-agents.mdx` and `action-guide-without-agents.mdx`.
>     - Expanded on creating custom actions with examples for existing and new tools in `custom_actions.mdx`.
>     - Provided examples for finding actions by use case in `usecase.mdx`.
>   - **File Renames and Restructuring**:
>     - Renamed `action-guide-with-agents.mdx` to `tools/action-guide-with-agents.mdx`.
>     - Renamed `action-guide-without-agents.mdx` to `tools/action-guide-without-agents.mdx`.
>     - Renamed `custom_actions.mdx` to `tools/custom_actions.mdx`.
>     - Renamed `usecase.mdx` to `tools/filter-actions.mdx`.
>     - Added `tools/what-are-tools.mdx` to explain tools and actions.
>   - **Formatting and Clarity**:
>     - Removed redundant titles in code blocks across multiple files.
>     - Improved descriptions and added tips for better understanding of custom actions and use cases.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=ComposioHQ%2Fcomposio&utm_source=github&utm_medium=referral)<sup> for ec4b6c00a426a67f4070c38c3781bdd3b4f7fe0e. It will automatically update as commits are pushed.</sup>

### CODE REVIEW

File: docs/patterns/actions/action-guide-faqs.mdx [hunk 1]
```diff
@@ -25,47 +25,12 @@ const tools = await toolset.getTools({actions: ['github_issues_create','github_c
 
 </Accordion>
```
Comment (shreysingla11) [inline docs/patterns/actions/action-guide-faqs.mdx:27]: The removal of the "How do I find an Action for my Use Case?" section might reduce the documentation's helpfulness. Consider keeping this section or integrating its content into other parts of the document to maintain comprehensive guidance for users.

File: docs/patterns/actions/action-guide-faqs.mdx [hunk 2]
```diff
@@ -25,47 +25,12 @@ const tools = await toolset.getTools({actions: ['github_issues_create','github_c
...
-</CodeGroup>
-
-</Accordion>
-
-
 <Accordion title="Can I filter and get Actions in an App based on tags?">
-
-<Tabs>
-<Tab title="Python">
+Yes, you can filter and get Actions in an App based on tags.
```
Comment (shreysingla11) [inline docs/patterns/actions/action-guide-faqs.mdx:29]: Great improvement in code formatting and clarity. Consider adding brief explanations for each code example to further enhance understanding for users.

File: docs/patterns/actions/action-guide-with-agents.mdx [hunk 3]
```diff
@@ -68,4 +68,4 @@ agent_executor.invoke({"input": task})
 </CodeGroup>
 
 This code demonstrates how to use LLMs to execute an action. 
-Composio supports multiple frameworks for creating Agents including LlamaIndex, CrewAI, Letta, Langchain, you can see it in the **Supported Frameworks** section.
+Composio supports multiple frameworks for creating Agents including LlamaIndex, CrewAI, Letta, LangChain, you can see it in the **Supported Frameworks** section.
```
Comment (shreysingla11) [inline docs/patterns/actions/action-guide-with-agents.mdx:71]: Good correction of "Langchain" to "LangChain". Consider adding brief descriptions of each mentioned framework (LlamaIndex, CrewAI, Letta, LangChain) to provide more context for users who might be unfamiliar with these tools.

File: docs/patterns/actions/action-guide-without-agents.mdx [hunk 4]
```diff
@@ -9,6 +9,58 @@ description: "Perform Actions with your account or on behalf of your users witho
 
 Composio lets you perform actions with your account or on behalf of your users. Feel free to check all the [tools](https://app.composio.dev/apps) we support. 
 
+### Get Parameters for an Action
```
Comment (shreysingla11) [inline docs/patterns/actions/action-guide-without-agents.mdx:12]: The new "Get Parameters for an Action" section is a valuable addition. Consider adding:
1. A brief explanation of why and when users might need to get parameters for an action.
2. A text-based alternative or summary of the video content for accessibility and quick reference.

File: docs/patterns/actions/usecase.mdx [hunk 5]
```diff
@@ -11,14 +11,14 @@ Find relevant actions for your use case by describing what you want to accomplis
 
 
 <CodeGroup>
-```Python Python Fetch Actions using use-case
+```Python Python
```
Comment (shreysingla11) [inline docs/patterns/actions/usecase.mdx:14]: The code formatting improvements in this file enhance readability. Consider adding more context or explanation around these use case examples to help users understand how to apply them effectively in their projects.

**Comment by shreysingla11** [comment]
Overall, this PR significantly improves the documentation for Composio actions. Here's a summary of the changes and some general suggestions:

1. The code formatting and clarity improvements across all files are excellent and enhance readability.

2. The new "Get Parameters for an Action" section in `action-guide-without-agents.mdx` is a valuable addition.

3. The description of Custom Actions in `custom_actions.mdx` is now clearer and more comprehensive.

General suggestions for further improvement:

1. Consider adding brief explanations for each code example to enhance understanding.

2. For the frameworks mentioned (LlamaIndex, CrewAI, Letta, LangChain), consider adding short descriptions to provide context for users unfamiliar with these tools.

3. In the new "Get Parameters for an Action" section, consider adding:
   - A brief explanation of why and when users might need to get parameters for an action.
   - A text-based summary of the video content for accessibility and quick reference.

4. For Custom Actions, consider adding one or two concrete examples to illustrate their potential applications.

5. In the use case examples, consider adding more context or explanation to help users understand how to apply them effectively in their projects.

6. Ensure consistent capitalization across the documentation (e.g., "Javascript" vs "JavaScript").

7. Consider adding cross-references (links) between related sections to help users navigate the documentation more easily.

8. Perform a final proofreading pass for any remaining typos or grammatical issues.

These changes significantly enhance the documentation's quality and usefulness. Great work on improving the user experience for working with Composio actions!

File: docs/patterns/actions/action-guide-faqs.mdx (hunk 1 above)
Comment (abhishekpatil4) [reply docs/patterns/actions/action-guide-faqs.mdx:27]: Yes, there is already an entire section on [Finding Actions by Use Case](https://docs.composio.dev/patterns/actions/usecase) so removed the accordion.

File: docs/patterns/actions/action-guide-faqs.mdx (hunk 2 above)
Comment (abhishekpatil4) [reply docs/patterns/actions/action-guide-faqs.mdx:29]: yes, added descriptions.

File: docs/patterns/actions/action-guide-with-agents.mdx (hunk 3 above)
Comment (abhishekpatil4) [reply docs/patterns/actions/action-guide-with-agents.mdx:71]: this section is related to actions & there are independent section explaining each famework so didn't repeat it again here.

File: docs/patterns/actions/action-guide-without-agents.mdx (hunk 4 above)
Comment (abhishekpatil4) [reply docs/patterns/actions/action-guide-without-agents.mdx:12]: Done

File: docs/patterns/actions/custom_actions.mdx [hunk 6]
```diff
@@ -7,12 +7,12 @@ description: "Guide to creating Custom Actions for existing tool or new tool"
...
 
-- Extend the functionality of existing tools by creating a custom action which utilises the existing authentication. (Fetch emails from Gmail)
-- Write a custom action code snippet to execute the custom action (Check if number is even or odd)
+- Extend the functionality of existing tools by creating a custom action which utilises the existing authentication. (e.g.,fetch emails from Gmail)
+- Create standalone custom actions for any functionality you need (e.g., data processing, calculations, or integrations with other services)
 
 
 <Tip>
-Custom Actions can be created for any existing tool in Composio, allowing you to tailor the functionality to your specific needs while leveraging the existing authentication.
+Custom Actions are flexible building blocks that let you create any functionality you need - from simple calculations to complex integrations. They work seamlessly with Composio's authentication system for existing tools, or can run independently for custom logic.
```
[the remark this answers is not in the document]
Comment (abhishekpatil4) [reply docs/patterns/actions/custom_actions.mdx:15]: Yes, I'm going to revamp this section.

File: docs/patterns/actions/usecase.mdx (hunk 5 above)
Comment (abhishekpatil4) [reply docs/patterns/actions/usecase.mdx:14]: Sure

**Pull request merged**